docs(tabs): improvements to storybook for tabs

jenndiaz opened the pull request:
<!--
  Note: Before sending a pull request, ensure there's an issue filed for the change to track the work.
   - Search for (issues)[https://github.com/adobe/spectrum-css/issues]
   - If there's no issue, please [file it](https://github.com/adobe/spectrum-css/issues/new/choose) and tag @pfulton or @castastrophe for feedback.
-->
## Description

This PR adds the [tabs overflow example](https://opensource.adobe.com/spectrum-css/tabs.html#basictabswithoverflow) to storybook and decreasing the number of stories from 26 to 3 using our "kitchen sink" approach. 

[Design Spec for Tab overflow](https://spectrum.adobe.com/page/tabs/#Tab-overflow)

<!-- Describe what you changed and link to relevant issue(s) (e.g., #000) -->

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps
<!--
  Include steps for the PR reviewer that explain how they should test your PR. Example test outline:
  1. Open the [storybook](url) for the button component:
    - [ ] Hover over the button and validate the new hover background color is applied
    - [x] Click the button and validate the new active background color is applied [@castastrophe]
-->

Within the storybook for tabs: @mdt2 

- [x] toggling controls can replicate all of the stories for tabs that current exist on main 
- [x] toggling controls works as expected 
- [x] [Chromatic is displaying](https://www.chromatic.com/build?appId=64762974a45b8bc5ca1705a2&number=1212) the "kitchen sink" tabs groups as expected 


### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [x] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## Screenshots

<!-- If applicable, add screenshots to show what you changed -->

## To-do list

<!-- Put an "x" to indicate you've done each of the following. Add/remove additional tasks, as needed. -->

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] I have tested these changes in Windows High Contrast mode.
- [x] If my change impacts **other components**, I have tested to make sure they don't break.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

pfulton reviewed (changes requested):
One thing that I noticed here is that the t-shirt sizing control doesn't seem to affect the overflow variants. Can we address that?

jenndiaz commented:
> One thing that I noticed here is that the t-shirt sizing control doesn't seem to affect the overflow variants. Can we address that?

I implemented t-shirt sizing but this introduced complexity with positioning off popover. Since we're more concerned about the tabs implementation and not verifying the popover position, I also removed the popover from the overflow stories.

mdt2 commented on components/tabs/stories/tabs-horizontal-compact.stories.js:
```diff
@@ -29,29 +28,60 @@ export default {
+	return html`
+		<div style="display: flex; flex-direction: column; gap: 2rem;">
+			${Template({
+          selectorStyle: {"width": "35px"},
```
I'm seeing some weirdness when toggling controls that I think is caused by this static selectorStyle. For example, if I turn on icons, the selector indicator looks too short:

![Screen Shot 2024-02-07 at 2 40 36 PM](https://github.com/adobe/spectrum-css/assets/23404383/51dd39cb-afef-4c33-a29c-5df6ce90d9dc)

jenndiaz replied:
Addressed this!

mdt2 commented on components/tabs/stories/tabs-horizontal-compact.stories.js:
```diff
@@ -29,29 +27,54 @@ export default {
+			${Template({
+        ...args
+			})}
+			${window.isChromatic() ?
+				Template({
+          ...args,
+          labelWithIcons: true
+				})
+			: null }
+      ${window.isChromatic() ?
+        Template({
+          ...args,
+          iconOnly: true
+        })
+				: null }
+      ${window.isChromatic() ?
+        Template({
+          ...args,
+          isEmphasized: true
+        })
+      : null }
+      ${window.isChromatic() ?
+        Template({
+          ...args,
+          isEmphasized: true,
+          labelWithIcons: true
+        })
+      : null }
+      ${window.isChromatic() ?
+        Template({
+          ...args,
+          isEmphasized: true,
+          iconOnly: true
+        })
+      : null }
```
This is just style preference, aka optional, but you could move these into a const and use lit's `when` here. Something like [Combobox](https://github.com/adobe/spectrum-css/blob/main/components/combobox/stories/combobox.stories.js), but that aligns with the new `window.isChromatic()` setup.

mdt2 reviewed (approved):
This is looking good to me! One optional suggestion.

castastrophe commented:
@jenndiaz This is looking better but I still feel like there are way too many separate pages for the tabs. When we eventually expand test coverage, we'd be running against 16+ different snapshots. Are there ways we can collate these down further? Maybe just horizontal variants + vertical variants?

castastrophe commented on components/tabs/stories/index.js:
```diff
@@ -1,6 +1,8 @@
 export const argTypes = {
   items: { table: { disable: true }},
   selectorStyle: { table: { disable: true }},
+  style: { table: { disable: true }},
```
Instead of style, can we use `customStyles` which is baked into the global storybook settings.

castastrophe commented on components/tabs/stories/tabs-horizontal.stories.js:
```diff
@@ -30,11 +28,110 @@ export default {
 
-export const Default = Template.bind({});
-Default.args = horizontalProps;
+const chromaticKitchenSink = ({
```
```suggestion
const HorizontalTabsGroup = ({
```

I've noticed a move toward `chromaticKitchenSink` but this name is awfully generic. If we want to export the set to use elsewhere, we'd have to rename it on entry. I also love when variable names are as specific as possible to make it clear without documentation.

castastrophe commented on components/tabs/stories/tabs-horizontal.stories.js:
```diff
@@ -30,11 +28,110 @@ export default {
+}) => {
+	return html`
+		<div style="display: flex; flex-direction: column; gap: 2rem;">
+    <h2>Horizontal Default</h2>
```
If we're going to label items in the snapshots, can we run them through typography? That protects us from unexpected environment changes (font-size, typography, etc) from influencing all our snapshots. See button for some examples.

castastrophe commented on components/tabs/stories/tabs-horizontal.stories.js:
```diff
@@ -30,11 +28,110 @@ export default {
+	return html`
+		<div style="display: flex; flex-direction: column; gap: 2rem;">
+    <h2>Horizontal Default</h2>
+			${Template({
+        ...args
+			})}
+			${Template({
+        ...args,
+        labelWithIcons: true
+       })}
+      ${Template({
+        ...args,
+        iconOnly: true
+        })}
```
Since these are repeated in the 3 contexts below, can we pull that out into its own function? Maybe those are named `TabsGroup`? Also, since this same set appears in the vertical set, maybe move that definition to index.js and export it for use in both files?

castastrophe commented on components/tabs/stories/template.js:
```diff
@@ -16,11 +17,115 @@ export const Template = ({
+		]
+	}
+
+	const overflowHtml = html`
```
I couldn't tell if overflowHtml is used elsewhere or if it's just returned below?  If just returned below, can we skip adding it to a variable at all and just wrap this in the if statement and return the HTML directly?

castastrophe commented on components/tabs/stories/tabs-horizontal.stories.js:
```diff
@@ -30,11 +30,57 @@ export default {
+      ${Typography({
+        semantics: "heading",
+        size: "s",
+        content: ["Horizontal Default"],
```
```suggestion
        content: ["Horizontal default"],
```

Would you update headings to use sentence case? Thank you!

castastrophe commented:
@jenndiaz Just looking through the baselines and they're looking good but the spacing is a little tight with the visuals and typography, do you think you could increase the gap? https://www.chromatic.com/test?appId=64762974a45b8bc5ca1705a2&id=65cf8cc9273596f853df88ce

castastrophe commented on components/tabs/stories/tabs-horizontal.stories.js:
```diff
@@ -30,11 +30,59 @@ export default {
+const overflowTabsGroup = ({
+  ...args
+}) => {
+  return html `
```
I can't really tell if this is working in the snapshots or not - can you add some more contextual data to the snapshot so we can tell at a glance if it's handling overflow correctly?
